src/peft/peft_model.py

```diff
@@ -650,6 +677,84 @@ def _split_kwargs(cls, kwargs: Dict[str, Any]):
 
         return hf_hub_download_kwargs, other_kwargs
 
+    def _update_offload(self, offload_index: Dict[Dict[str:str]], adapters_weights: Dict[str : torch.tensor]):
+        """
+        Update the offload_index and safetensors files for loading and mergine PeftModels with disk-offloaded modules.
+
+        Args:
+            offload_index (Dict[str: str]):
+                Dictionary of disk-offloaded modules with their metadata and safetensors filenames
+            adapters_weights (Dict[str: torch.tensor]):
+                Dictionary of Peft adapter module names and weights
+        """
+
+        prefix = "base_model.model."
```

Comment:
I wonder: Should we create a shallow copy of `offload_index`? This would be to avoid having a function that both returns a value and mutates an input argument. Alternatively, we could make the function not return anything but mutate the `offload_index`. So below we would have:

```diff
- offload_index = self._update_offload(offload_index, adapters_weights)
+ self._update_offload(offload_index, adapters_weights)
```

I personally prefer the latter suggestion. WDYT?

Reply:
I think the second should be cleaner, and you are right that we don't really need to return the `offload_index` anyways